modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java

```diff
@@ -362,6 +367,44 @@ public final boolean isOpaque() {
         return false;
     }
 
+    // we need to know the JavaFX screen of the current AWT graphcisConfiguration
+    private Screen findScreen(GraphicsConfiguration graphicsConfiguration) {
+        Rectangle awtBounds = graphicsConfiguration.getBounds();
+        AffineTransform awtScales = graphicsConfiguration.getDefaultTransform();
+        for (Screen screen : Screen.getScreens()) {
+            if ((Math.abs(screen.getPlatformX() - awtBounds.getX()) < 2.) &&
+                    (Math.abs(screen.getPlatformY() - awtBounds.getY()) < 2.) &&
+                    (Math.abs(screen.getPlatformWidth() - awtScales.getScaleX() * awtBounds.getWidth()) < 2.) &&
+                    (Math.abs(screen.getPlatformHeight() - awtScales.getScaleY() * awtBounds.getHeight()) < 2.)) {
+                return screen;
+            }
+        }
+        return null;
+    }
+
+    private Dimension2D convertSwingToFxPixel(GraphicsConfiguration g, float wx, float wy) {
+        float newx, newy;
+        Screen screen = findScreen(getGraphicsConfiguration());
+        if (screen != null) {
+            AffineTransform awtScales = getGraphicsConfiguration().getDefaultTransform();
+            float pScaleX = screen.getPlatformScaleX();
+            float pScaleY = screen.getPlatformScaleY();
+            float sx = screen.getX();
+            float sy = screen.getY();
+            float awtScaleX = (float)awtScales.getScaleX();
+            float awtScaleY = (float)awtScales.getScaleY();
+
+            float px = screen.getPlatformX();
+            float py = screen.getPlatformY();
+            newx = sx + (wx - px) * awtScaleX / pScaleX;
+            newy = sy + (wy - py) * awtScaleY / pScaleY;
```

Comment:
`px` and `py` are already (scaled) "FX" coordinates, so I'm not sure AWT scale should be applied to that. I might be missing something here.

Reply:
Without this correction, there is an offset on Linux (where awtScaleX = 1 and pScaleX =2  for example).
TBH, I think the relations between the different scale factors should be written out clearly somewhere, as I think there are a number of places where we face platform-specific differences.